content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/_index.md

```diff
@@ -0,0 +1,335 @@
+
+#### Outputs
+
+The `PythonInstrumentedTestRunner` returns the same outputs as the `TestRunnerWithCoverage`.
+
+### NativeInstrumentedTestRunner
+
+The `NativeInstrumentedTestRunner` derives from the `TestRunnerWithCoverage` and implements the `PayloadExtractor` to handle the extraction of GTest test run artifacts and the JUnit coverage artifacts.
+
+#### Inputs
+
+The `NativeInstrumentedTestRunner` accepts the same inputs as the `TestRunnerWithCoverage`.
+
+#### Process
+
+The `NativeInstrumentedTestRunner` extracts test run payloads from GTest XML files and coverage payloads from JUnit coverage artifacts produced by OpenCppCoverage.
+
+#### Outputs
+
+The `NativeInstrumentedTestRunner` returns the same outputs as the `TestRunnerWithCoverage`.
+
+
+### Native sharded test runners
+
+The `NativeShardedRegularTestRunner` and `NativeShardedInstrumentedTestRunner` are an optimization that split opted in test targets into shards to distribute over the available hardware cores for increased performance. They do this transparently by presenting a similar interface to the standard test runners. They do this by feeding the sharded test target `JobInfos` to said standard `NativeRegularTestRunner` and `NativeInstrumentedTestRunner` test runners (where there will potentially be multiple `JobInfos` per test target) before consolidating the output of said standard test runners into one `Job` per test target. Both of the sharded test runners derive from `NativeShardedTestRunnerBase` (not shown in the diagram above) and implement the `ConsolidateSubJobs` function. 
+
+
+#### Inputs
+
+The native sharded test runners accepts `ShardedTestJobInfos` as their input where each `ShardedTestJobInfo` contains information about the parent job that is presented to the user and the sharded sub jobs that are fed to the standard test runners.
+
+#### Process
+
+The native sharded test runners wait until the sharded sub jobs of a given parent job are complete before presenting the appropriate notification to the user. Once all sharded sub jobs are complete, the native sharded test runners consolidate the sharded sub jobs back into the parent job and present these parent jobs to the user, as if the sharding never occured.
+
+#### Outputs
+
+The native sharded test runners returns the same outputs as the  `NativeRegularTestRunner` and `NativeInstrumentedTestRunner`.
+
+#### FAQ
+
+##### Why are there no Python sharded test runners like the native shardedtest runner counterparts?
+
+As the python tests cannot be arbitrarily executed in parallel at the test level, there is no need for sharded Python test runners.S
```

Comment:
update: Random `S` at the end can be removed 🙂

Reply:
Fixed 👍